<!DOCTYPE html>
<html>
	<head>
		<title>Asyanc and Await</title>

		<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.4/css/all.min.css"
		integrity="sha512-1ycn6IcaQQ40/MKBW2W4Rhis/DbILU74C1vSrLJxCq57o941Ym01SwNsOMqvEBFlcgUa6xLiPY/NS5R+E6ztJQ=="
		crossorigin="anonymous" referrerpolicy="no-referrer" />

		<style>
			body{
				height: 100vh;

				display: flex;
				justify-content: center;
				align-items: center;
			}

					/* min   default  max */
			/* clamp(200px,75%,300px); */
			/* .container{
				width:clamp(350px,90%,600px);
				background-color: #f4f4f4;
				padding: 10px;
			} */


			.container{
				width: 300px;
				background-color: #f4f4f4;
				padding: 10px;
			}

			.container label{
				display: block;
			}

			.container .form-control{
				width: 100%;
				margin-bottom: 10px;
			}

		</style>
	</head>
	<body>
		

		<script src="https://code.jquery.com/jquery-3.7.1.min.js" type="text/javascript"></script>
		<script src="https://unpkg.com/axios/dist/axios.min.js"></script>

		<script type="text/javascript">
			
			function myfunone(){
				return "Hello sir . i am myfunone";
			};
			console.log(myfunone());

			// async = synchronous

			async function myfuntwo(){
				return "Hello sir , i am myfuntwo";
			};

			console.log(myfuntwo());  // PromiseÂ {<fulfilled>: 'Hello sir , i am myfuntwo'} 

			myfuntwo().then((response)=>{
				console.log(response);  // Hello sir , i am myfuntwo
			});


			async function myfunthree(){
												// two parameter
				const setpromise = new Promise((resolve,reject)=>{
					setTimeout(()=>resolve("Hello World"),3000);
				});  // distribute
				const response = await setpromise;
				return response
				// await keyword 
			}
			console.log(myfunthree());  // not working , result = no
			myfunthree().then((response)=>console.log(response));


		

			async function myfunfour(){

				const setpromise = new Promise((resolve,reject)=>{
					setTimeout(()=>resolve("Hello Earth"),3000);
				});

				const error = true;  // false

				if(!error){
					const response = await setpromise;
					return response;
				}else{
					await Promise.reject(new Error("Something went wrong !!!"));
				}

			}

			// console.log(myfunfour());
			myfunfour()
			.then(response=>console.log(response))
			.catch((err)=>console.log(err));
			
			
		</script>
	</body>
</html>

<!-- .fetch(request)
.then(promise)
.then(response) -->